src/main/java/edu/wpi/grip/ui/OperationView.java

```diff
@@ -88,5 +88,28 @@ public void initialize(URL url, ResourceBundle resourceBundle) {
             Step step = new Step(this.eventBus, this.operation);
             this.eventBus.post(new StepAddedEvent(step));
         });
+
+
+        // Ensures that when this element is hidden that it also removes its size calculations
+        this.managedProperty().bind(this.visibleProperty());
+    }
+
+
+    /**
+     * Uses the search string from the palette and allows this element to determine if it should be visible
+     *
+     * @param searchString The search string that should be used to determine visibility
+     */
+    public void determineVisibility(final String searchString) {
```

Comment:
Its an, I act on myself, instead of an, I am acted upon design pattern. Better encapsulation.